internal/controllers/vgmanager/lsblk/lsblk.go

```diff
@@ -125,26 +136,72 @@ func (lsblk *HostLSBLK) IsUsableLoopDev(b BlockDevice) (bool, error) {
 	return true, nil
 }
 
-// HasBindMounts checks for bind mounts and returns mount point for a device by parsing `proc/1/mountinfo`.
-// HostPID should be set to true inside the POD spec to get details of host's mount points inside `proc/1/mountinfo`.
-func (lsblk *HostLSBLK) HasBindMounts(b BlockDevice) (bool, string, error) {
-	data, err := os.ReadFile(lsblk.mountInfo)
-	if err != nil {
-		return false, "", fmt.Errorf("failed to read file %s: %v", lsblk.mountInfo, err)
-	}
+type BlockDeviceInfos map[string]BlockDeviceInfo
+
+type BlockDeviceInfo struct {
+	HasBindMounts   bool
+	IsUsableLoopDev bool
+}
 
-	mountString := string(data)
-	for _, mountInfo := range strings.Split(mountString, "\n") {
-		if strings.Contains(mountInfo, b.KName) {
-			mountInfoList := strings.Split(mountInfo, " ")
-			if len(mountInfoList) >= 10 {
-				// device source is 4th field for bind mounts and 10th for regular mounts
-				if mountInfoList[3] == fmt.Sprintf("/%s", filepath.Base(b.KName)) || mountInfoList[9] == b.KName {
-					return true, mountInfoList[4], nil
-				}
+func flattenedBlockDevices(bs []BlockDevice) map[string]BlockDevice {
+	flattened := make(map[string]BlockDevice, len(bs))
+	for _, b := range bs {
+		flattened[b.KName] = b
+		if b.HasChildren() {
+			for k, v := range flattenedBlockDevices(b.Children) {
+				flattened[k] = v
 			}
 		}
 	}
+	return flattened
+}
+
+func (lsblk *HostLSBLK) BlockDeviceInfos(bs []BlockDevice) (BlockDeviceInfos, error) {
+	flattenedMap := flattenedBlockDevices(bs)
+
+	blockDeviceInfos := make(BlockDeviceInfos)
+	// file, err := os.Open(lsblk.mountInfo)
```

Comment:
I guess these are just some leftover code that can be removed

Reply:
We dont do bind mount checks anymore in microlvms. Thats important to notice if we run into an issue later. Removed the code for now.